### PR TITLE
Improve netpol validation retry

### PR DESCRIPTION
**Pull request by stbenjam**
#### What type of PR is this?

/kind cleanup
/kind flake

#### What this PR does / why we need it:

ValidateOrFail only retries twice, and assumes that the time it took to
test the first time, was enough to wait to try again.  This refactors
the function to poll instead.

This also removes a now unnecessary 3 second wait in the policy
deletion test.

We're seeing in OpenShift OVN, the test that we should stop enforcing
policies after they are deleted fails occasionally as the implicit wait in
ValidateOrFail isn't long enough to ensure the policies are actually
deleted.

#### Which issue(s) this PR fixes:

Test: `[sig-network] Netpol [LinuxOnly] NetworkPolicy between server and client should stop enforcing policies after they are deleted`

https://testgrid.k8s.io/redhat-openshift-ocp-release-4.9-informing#periodic-ci-openshift-release-master-nightly-4.9-e2e-metal-ipi-ovn-dualstack

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig network

### CODE REVIEW

**Comment by astoycos** [comment]
@jayunit100 Can you take a look at this?

File: test/e2e/network/netpol/test_helper.go [hunk 1]
```diff
@@ -68,23 +69,25 @@ func UpdatePolicy(k8s *kubeManager, policy *networkingv1.NetworkPolicy, namespac
 func ValidateOrFail(k8s *kubeManager, model *Model, testCase *TestCase) {
 	ginkgo.By("Validating reachability matrix...")
 
-	// 1st try
-	ginkgo.By("Validating reachability matrix... (FIRST TRY)")
-	ProbePodToPodConnectivity(k8s, model, testCase)
-	// 2nd try, in case first one failed
-	if _, wrong, _, _ := testCase.Reachability.Summary(ignoreLoopback); wrong != 0 {
-		framework.Logf("failed first probe %d wrong results ... retrying (SECOND TRY)", wrong)
+	err := wait.PollImmediate(3*time.Second, 15*time.Second, func() (done bool, err error) {
```
Comment (mattfenwick) [inline test/e2e/network/netpol/test_helper.go:72]: How did you arrive at 3 seconds for interval, and 15 seconds for timeout?  

Looking at [a recent test run](https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-ubuntu-gce-network-policies/1409703242257928192/build-log.txt), a `ProbePodToPodConnectivity` probe can take ~ 2 minutes.  With the timeout set to 15 seconds, wouldn't this change effectively stop any retries from happening?  I'm not sure that's what we want.

Comment (stbenjam) [reply test/e2e/network/netpol/test_helper.go:72]: An educated guess from @danwinship. I'll take a closer look, thanks!

**Comment by stbenjam** [comment]
/close

Not needed

**Pull request closed without merging**